src/streaming/streamingclient.py

```diff
@@ -0,0 +1,178 @@
+                on_close = on_close, 
+                on_open = on_open):
+        """Constructor for Streaming Client
+
+        :param access_token: access token which authorizes all requests and links them to your
+                             account. Generated on the settings page of your account dashboard
+                             on Rev.ai.
+        :param config: an AudioConfig object containing audio information. 
+                       See AudioConfig.py for more information
+        :param version (optional): version of the streaming speech to text as string
+        :param on_message (optional): function to be called when recieving a message from the server
+        :param on_error (optional): function to be called when recieving an error from the server
+        :param on_close (optional): function to be called when the websocket closes
+        :param on_open (optional): function to be called when the websocket and thread starts successfully
+        """
+        if not access_token:
+            raise ValueError('access_token must be provided')
+
+        if not config:
+            raise ValueError('config must be provided')
+
+
+        self.access_token = access_token
+        self.config = config
+        self.base_url = \
+                'wss://api.rev.ai/speechtotext/{}/stream'.format(version)
+        self.on_error = on_error
+        self.on_close = on_close
+        self.on_open = on_open
+        self.thread = SpeechThread
+        self.client = websocket.WebSocket(enable_multithread = True, 
+            on_error = self.on_error, 
+            on_close = self.on_close, 
+            on_open = self.on_open
+            )
+
+    def start(self):
+        """Function to connect the websocket to the URL and start the response thread
+        """
+        url = self.base_url + '?access_token={}'.format(self.access_token) \
+              + '&content_type={}'.format(self.config.get_content_type_string()
+                )
+        try:
+            self.client.connect(url)
+            self.on_open()
+        except Exception as e:
+            self.client.abort()
+            self.on_error(e)
+
+    def end(self):
+        """Function to end the streaming service, close the websocket.
+        """
+        self.client.abort()
+
+    def send_data_as_gen(self, generator):
+        """Function to send binary audio data from a generator
+
+        :param generator: generator object that yields binary audio data
+        """
+        if not generator:
+            raise ValueError('generator must be provided')
+
+        if hasattr(self, 'request_thread'):
+            if self.request_thread.isAlive():
+                raise ValueError("""Data is still being sent and will interfere 
+                    with the responses.""")
+
+        self.request_thread = self.thread(self.RequestSending, [generator])
+        self.request_thread.start()
+
+    def send_data_as_filepath(self, filepath):
```

Comment:
I don't believe that this method fit in this class. The responsibility of this class is to send and receive data from rev.ai. This filepath method gives the client more responsibility and makes it harder for us in the future. 

For example, currently if we're passed a file which we don't have permission to read the read operation will error and it will look like the client is erroring. In general we should avoid interacting with files as it puts the responsibility of handling a lot of edge cases on us.

Customers expect that if we say we're going to take responsibility for something (in this case turning their file into a generator) that we will gracefully handle any errors that occur during that process. This method currently doesn't do that and actually handling it way more work than we want to do